### PR TITLE
AJAX scheduler: option to hide unapproved classes.

### PR DESCRIPTION
**Pull request by uakfdotb**
This mimics the text and filtering strategy from the old AJAX scheduler.

### CODE REVIEW

**Comment by uakfdotb** [comment]
ref #1801

File: esp/public/media/scripts/ajaxschedulingmodule/ESP/Sections.js [hunk 1]
```diff
@@ -47,6 +48,9 @@ function Sections(sections_data, section_details_data, teacher_data, scheduleAss
         }.bind(this));
         return result;
     }.bind(this);
+    this.filter.classHideUnapproved.valid = function(a) {
+        return a.status === 10;
```
Comment (uakfdotb) [inline esp/public/media/scripts/ajaxschedulingmodule/ESP/Sections.js]: Not sure if there's a better way to get the approved class status number.

Comment (luac) [inline esp/public/media/scripts/ajaxschedulingmodule/ESP/Sections.js]: status > 0 is used in a couple of places, notably ClassSubject.isAccepted():
https://github.com/learning-unlimited/ESP-Website/blob/main/esp/esp/program/models/class_.py#L1759

**Comment by mgersh** [comment]
I'm just going to merge this so people can look at everything over at ajax-spike.

**Pull request merged**